### PR TITLE
add xchain-terra 

### PR DESCRIPTION
**Pull request by mikewyszinski**

### CODE REVIEW

**Review by veado — CHANGES_REQUESTED** [review]
Nice job! Just few change requests ... 

Pls merge latest master again.

File: packages/xchain-terra/src/client.ts [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+}
+export type TerraClientParams = {
+  explorerURL?: string
+  explorerAddressURL?: string
+  explorerTxURL?: string
+  cosmosAPIURL?: string
+  ChainID?: string
+}
+
+const DECIMALS = 6
```
[the remark this answers is not in the document]
Comment (mikewyszinski) [reply packages/xchain-terra/src/client.ts:71]: yes -> https://pkg.go.dev/github.com/terra-money/core@v0.5.14/types/assets#pkg-types

File: packages/xchain-terra/src/client.ts [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    if (asset.chain === Chain.Terra && asset.symbol === 'LUNA' && asset.ticker === 'LUNA') {
+      amountToSend = {
+        uluna: `${amount.amount().toFixed()}`,
+      }
+    } else if (asset.chain === Chain.Terra && asset.symbol === 'UST' && asset.ticker === 'UST') {
+      amountToSend = {
+        uusd: `${amount.amount().toFixed()}`,
+      }
+    } else {
+      throw new Error('Only LUNA or UST transfers are currently supported on terra')
```
[the remark this answers is not in the document]
Comment (mikewyszinski) [reply packages/xchain-terra/src/client.ts:198]: yeah wasn't sure about this, glad you brought it up, i believe TC will only have UST/LUNA pools to start, but i'm going to remove those check and support all native terra xfers

File: packages/xchain-terra/src/client.ts [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+    }
+    const send = new MsgSend(wallet.key.accAddress, recipient, amountToSend)
+    // console.log(send.toJSON())
```
[the remark this answers is not in the document]
Comment (mikewyszinski) [reply packages/xchain-terra/src/client.ts:201]: done

File: packages/xchain-terra/__tests__/client.test.ts [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+//   nock(url)
+//     .get(`/tx_search`)
+//     .twice()
+//     .query((_) => true)
+//     .reply(200, result)
+// }
+
+// const assertTxHashGet = (url: string, hash: string, result: TxResponse): void => {
+//   nock(url).get(`/txs/${hash}`).reply(200, result)
+// }
```
[the remark this answers is not in the document]
Comment (mikewyszinski) [reply packages/xchain-terra/__tests__/client.test.ts:51]: i left this commented code because i need to add unit tests for other methods....still a WIP

**Pull request merged**